cli/src/action/keygen.rs

```diff
@@ -116,9 +117,18 @@ pub fn create_key_pair(
     private_key_path: PathBuf,
     public_key_path: PathBuf,
     force_create: bool,
+    skip_create: bool,
     change_permissions: bool,
-) -> Result<Vec<u8>, CliError> {
+) -> Result<(), CliError> {
     if !force_create {
+        if private_key_path.exists() && public_key_path.exists() && skip_create {
+            info!(
+                "Skipping, key already exists: {}",
+                private_key_path.display()
+            );
+            return Ok(());
+        }
+
         if private_key_path.exists() {
             return Err(CliError::EnvironmentError(format!(
                 "File already exists: {:?}",
```

Comment:
So if a user specifies `--skip` (for which the documentation says `Skip generating the files if they exist.`) the keygen command would still abort with the error `File already exists: ...` if only one of the keypair files exists.

Aborting makes sense because the files are a pair, and `--skip` should only skip if both exist. I just think the message in combination with `--skip` becomes contextually confusing. Maybe there should be a custom error that explains this when `skip_create` is true, but the files exist?

Reply:
Updated to check each case and return errors in the case of one file missing.